tests/serialization/test_dag_serialization.py

```diff
@@ -1174,6 +1175,24 @@ def test_operator_deserialize_old_names(self):
         serialized_op = SerializedBaseOperator.deserialize_operator(blob)
         assert serialized_op.downstream_task_ids == {'foo'}
 
+    def test_task_resources(self):
+        """
+        Test TaskGroup serialization/deserialization.
+        """
+        from airflow.operators.dummy import DummyOperator
+
+        execution_date = datetime(2020, 1, 1)
+        task_id = 'task1'
+        with DAG("test_task_resources", start_date=execution_date) as dag:
+            task = DummyOperator(task_id=task_id, resources={"cpus": 0.1, "ram": 2048})
+
+        SerializedDAG.validate_schema(SerializedDAG.to_dict(dag))
+
```

Comment:
Please can you add a test here that checks the json/dict structure, not just the end-to-end. (This is important for making sure that we can continue to load the serialized blobs we store in the DB.)

Reply:
> checks the json/dict structure

@ashb  do you mean check the json/dict structure of the task.resource? or the whole serialized dag json/dict?

Reply:
That the resources part of the task is correctly represented -- crucially that it's tested in this file, so that it's more obvious if someone makes a breaking change to how this is serialized in the future.

Reply:
This is sort of what I was thinking of https://github.com/apache/airflow/blob/bff0b101388dc2a7c0fdf550540d38240b020181/tests/serialization/test_dag_serialization.py#L1676-L1726

Reply:
thanks.